src/core/types/secret_bytes.rs

```diff
@@ -53,6 +53,12 @@ macro_rules! decl_secret_bytes {
                     .map(|secret_magic| $struct_name { secret_magic })
             }
 
+           impl $struct_name {
+                pub fn to_bytes(&self) -> &[u8] {
+                    &self.secret_magic.0.as_slice()
```

Comment:
please change `[< $struct_name:snake _to_bytes >](bytes: &$struct_name) ` below to use this method